android/src/main/java/com/rnuxcam/rnuxcam/UXCamModule.java

```diff
@@ -0,0 +1,85 @@
+package com.rnuxcam.rnuxcam;
+
+import com.uxcam.UXCam;
+
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.ReadableMapKeySetIterator;
+
+import java.util.HashMap;
+
+public class UXCamModule extends ReactContextBaseJavaModule {
+
+	public UXCamModule(ReactApplicationContext reactContext) {
+		super(reactContext);
+	}
+
+    @Override
+    public String getName() {
+        return "RNUXCam";
+    }
+
+    @SuppressWarnings("unused")
+    @ReactMethod
+	public void startWithKey(String key) {
+		UXCam.startApplicationWithKeyForCordova(getCurrentActivity(), key);
+	}
+
+	@SuppressWarnings("unused")
+	@ReactMethod
+	public void tagScreenName(String screenName) {
+		UXCam.tagScreenName(screenName);
+	}
+
+	@SuppressWarnings("unused")
+	@ReactMethod
+	public void tagUserName(String userName) {
+		UXCam.tagUsersName(userName);
+	}
+
+	@SuppressWarnings("unused")
+	@ReactMethod
+	public void addTag(String tag, ReadableMap properties) {
+		HashMap<String, String> map = new HashMap<String, String>();
+
+		ReadableMapKeySetIterator iterator = properties.keySetIterator();
+		while (iterator.hasNextKey()) {
+			String key = iterator.nextKey();
+			String value = properties.getString(key);
+			map.put(key, value);
+		}
+		UXCam.addTagWithProperties(tag, map);
+	}
+
+	@SuppressWarnings("unused")
+	@ReactMethod
+	public void markUserAsFavorite() {
+		UXCam.markSessionAsFavorite();
+	}
+
+	@SuppressWarnings("unused")
+	@ReactMethod
+	public String urlForCurrentUser() {
```

Comment:
I believe all bridge methods must be void (due to the async nature of react-native). Let's change this to return a Promise instead!

> To expose a method to JavaScript a Java method must be annotated using @ReactMethod. The return type of bridge methods is always void. React Native bridge is asynchronous, so the only way to pass a result to JavaScript is by using callbacks or emitting events (see below).

http://facebook.github.io/react-native/docs/native-modules-android.html#promises